main/wifi.c

```diff
@@ -235,8 +235,12 @@ static void wifi_ctrl(void* _param) {
           if (rxp.data[1] == 0) {
             wifi_init_sta(ssid, key);
           } else {
-            wifi_init_softap(ssid, key);
-          }          
+            if (strlen(key) < 8) {
```

Comment:
If we only check the length so it must be longer than 7, we can never set up an AP with open security. 

Inside the wifi_init_softap() the authentication mode is set to open if the password is longer than 0. 
I would say we should check the invalid range. 
```suggestion
            if (0 < strlen(key) && strlen(key) < 8) {
```

Reply:
Implemented and tested your suggestion